File: lib/onchain/scripts/js/gen-account.mjs
import fs from 'fs';
import {privateToAddress} from 'ethereumjs-util';
import {generateECDSAPrivateKey} from '../../../soda-sdk/js/crypto.mjs';

const envFileName = process.argv[2] || '.env';

// Generate ECDSA private key
const privateKey = generateECDSAPrivateKey().toString('hex');

// Create account address from private key
const address = privateToAddress(Buffer.from(privateKey, 'hex')).toString('hex');

console.log("Account address: ", address);

// Write the data to a .env file
const line = `export SIGNING_KEY='0x${privateKey}'\n`;

fs.writeFile(envFileName, line, (err) => {
  if (err) throw err;
});

File: lib/onchain/scripts/python/gen_account.py
from eth_account import Account
import argparse

parser = argparse.ArgumentParser()
parser.add_argument('env_file_name', help='Name of the env file')
args = parser.parse_args()

account = Account.create()

print("Account address: ", account.address)

# Clear the .env file and write the new signing key to the .env file
if args.env_file_name:
    with open(args.env_file_name, 'w') as f:
        f.write(f"export SIGNING_KEY='{account._private_key.hex()}'\n")
else:
    with open('.env', 'w') as f:
        f.write(f"export SIGNING_KEY='{account._private_key.hex()}'\n")